.github/actions/detect-exposed-secrets/exposed_secrets_detection/src/cli.py

```diff
@@ -0,0 +1,176 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+import argparse
+
+# Local
+from .custom_exceptions import (
+    GitWorkflowFilesNotFoundError,
+    ExposedSecretsError,
+    MissingArgumentsError,
+)
+from .utils import (
+    find_all_yaml_files_in_directory,
+    find_exposed_github_secrets_in_env_block,
+    find_exposed_github_secrets_in_job,
+    get_workflow_trigger_conditions,
+    is_git_workflow_file,
+    load_workflow_file,
+    workflow_requires_manual_trigger,
+)
+
+
+def create_parser() -> argparse.ArgumentParser:
+    """
+    Generates a parser for the CLI.
+
+    Returns
+    -------
+    parser: argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser(
+        prog="Exposed-Secrets-Detection",
+        description="Attempts to find any potentially exposed secrets, and reports them",
+    )
+
+    # User can pass in a directory or file, hence they both have "required=False", but one must be chosen
+    parser.add_argument(
+        "-d",
+        "--dir",
+        type=str,
+        help="Directory to read Git workflow files from",
+        required=False,
+    )
+    parser.add_argument(
+        "-f",
+        "--file",
+        type=str,
+        help="Path to the Git workflow file to read",
+        required=False,
+    )
+
+    return parser
+
+
+def run_validation(args: dict):
+    """
+    Runs validation steps to detect potentially-exposed secrets. In the process, this function also alidates
+    that the user has provided at least "--dir" or "--file" as inputs
+
+    Inputs
+    ------
+    args: dict
+        Dictionary of the user's arguments
+    """
+    read_dir = args.get("dir")
+    file_path = args.get("file")
+
+    if not read_dir and not file_path:
```

Comment:
Nit: Since nothing is catching this error, it might be better to print a message end exit(1)

Reply:
Similar comment for the below errors

Reply:
Changed from exception to: `print()` + `sys.exit(1)`